docs/docs/tutorials/security/authorization.mdx

```diff
@@ -0,0 +1,1064 @@
+{
+  "resource": "https://your-server.com/mcp",
+  "authorization_servers": ["https://auth.your-server.com"],
+  "scopes_supported": ["mcp:tools", "mcp:resources"]
+}
+```
+
+You can see a more comprehensive example in [RFC 9278 Section 3.2](https://datatracker.ietf.org/doc/html/rfc9728#name-protected-resource-metadata-r).
+
+</Step>
+
+<Step title="Authorization Server Discovery">
+Next, the client discovers what the authorization server can do by fetching its metadata. If the PRM document lists more than one authorization server, the client can decide which one to use. 
+
+With an authorization server selected, the client will then construct a standard metadata URI and issue a request to the [OpenID Connect (OIDC) Discovery](https://openid.net/specs/openid-connect-discovery-1_0.html) or [OAuth 2.0 Auth Server Metadata](https://datatracker.ietf.org/doc/html/rfc8414) endpoints (depending on authorization server support)
+ and retrieve another set of metadata properties that will allow it to know the endpoints it needs to complete the authorization flow.
+
+```json
+{
+  "issuer": "https://auth.your-server.com",
+  "authorization_endpoint": "https://auth.your-server.com/authorize",
+  "token_endpoint": "https://auth.your-server.com/token",
+  "registration_endpoint": "https://auth.your-server.com/register"
+}
+```
+
+</Step>
+
+<Step title="Client Registration">
+With all the key metadata out of the way, the client now needs to make sure that it's registered with the authorization server. This can be done in two ways.
+
+First, the client can be **pre-registered** with a given authorization server, in which case it can have embedded client registration information that it uses to complete the authorization flow. Alternatively, the client can use **Dynamic Client Registration** (DCR) to dynamically register itself with the authorization server. The latter scenario requires the authorization server to support DCR.
+
+If the authorization server does support, the client will send a request to the `registration_endpoint` with its information:
+
+```json
+{
+  "client_name": "My MCP Client",
+  "redirect_uris": ["http://localhost:3000/callback"],
+  "grant_types": ["authorization_code", "refresh_token"],
+  "response_types": ["code"]
+}
+```
+
+If the registration succeeds, the authorization server will return a JSON blob with client registration information, including the original registration metadata as well as `client_id`, `client_secret` (if applicable), `client_id_issued_at` (optional), and `client_secret_expires_at` (if `client_secret` is used).
+
+<Tip>
+**No DCR or Pre-Registration**
+
+In case a MCP client connects to a MCP server that doesn't use an authorization server that supports DCR and the client is not pre-registered with said authorization server, it's the responsibility of the client developer to provide an affordance for the end-user to enter client information manually.
+</Tip>
+
+</Step>
+
+<Step title="User Authorization">
```

Comment:
This step combines the `/authorize` endpoint, the OAuth auth code callback handling, and the `/token` endpoint. If the audience isn't familiar with OAuth, this could be broken out into a few more steps.

Reply:
Good point, I can fix this.